### PR TITLE
browser: add webContents.debugger api

### LINKED ISSUE #4002: Support Debugger API for Webview
I see Chrome have Debugger API, it's very useful to interact with webview without remote debugging protocol.
I hope Electronjs have this feature.
Thanks.

### PR DESCRIPTION
**Pull request by deepak1556**
For #4002 

- [x] Add tests

@zcbenz is this api implementation fine or should debugger be made a separate module ?

### CODE REVIEW

File: atom/browser/api/atom_api_debugger.cc [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  if (agent_host_->IsAttached()) {
+    args->ThrowError("Another debugger is already attached to this target");
+    return;
+  }
+
+  agent_host_->AttachClient(this);
+}
+
+void Debugger::Detach() {
+  agent_host_->DetachClient();
```
Comment (zcbenz) [inline atom/browser/api/atom_api_debugger.cc]: Should check `agent_host_` here.

File: docs/api/web-contents.md [hunk 2]
```diff
@@ -833,3 +833,66 @@ Get the `WebContents` of DevTools for this `WebContents`.
...
+      win.webContents.debugger.detach();
+  }
+})
+
+win.webContents.debugger.sendCommand("Network.enable");
+```
+
+#### `webContents.debugger.attach([protocolVersion])`
+
+* `protocolVersion` String - Required debugging protocol version.
```
Comment (zcbenz) [inline docs/api/web-contents.md]: `Required` => `Requested`.

File: docs/api/web-contents.md [hunk 3]
```diff
@@ -833,3 +833,66 @@ Get the `WebContents` of DevTools for this `WebContents`.
...
+* `method` String - Method name, should be one of the methods defined by the
+   remote debugging protocol.
+* `commandParams` Object - JSON object with request parameters.
+* `callback` Function - Response
+  * `result` Object - Response defined by the 'returns' attribute of
+     the command description in the remote debugging protocol.
+
+Send given command to the debugging target.
+
+#### `webContents.debugger.onDetach(callback)`
```
Comment (zcbenz) [inline docs/api/web-contents.md]: The traditional style of Electron API is to make `debugger` a EventEmitter.

**Comment by zcbenz** [comment]
> @zcbenz is this api implementation fine or should debugger be made a separate module ?

Making it inside Electron looks good to me.

**Comment by deepak1556** [comment]
Have made the changes, thanks!

File: atom/browser/api/lib/web-contents.js [hunk 4]
```diff
@@ -70,6 +70,9 @@ let wrapWebContents = function(webContents) {
   var controller, method, name, ref1;
   webContents.__proto__ = EventEmitter.prototype;
 
+  // webContents.debugger is an EventEmitter.
+  webContents.debugger.__proto__ = EventEmitter.prototype;
```
Comment (zcbenz) [inline atom/browser/api/lib/web-contents.js]: This is going to initialize the `debugger` for every `webContents` no matter whether users use it, this doesn't take much resources but is not good. Can you use the `_setWrap` trick to do make it inherit `EventEmitter`?

Comment (deepak1556) [inline atom/browser/api/lib/web-contents.js]: Fixed, thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**